packages/scheduler/src/forks/SchedulerHostConfig.default.js

```diff
@@ -271,7 +261,12 @@ if (
     };
     rAFTimeoutID = setTimeout(onTimeout, frameLength * 3);
 
-    if (prevRAFTime !== -1) {
+    if (
+      prevRAFTime !== -1 &&
+      // Make sure this rAF time is different from the previous one. This check
+      // could fail if two rAFs fire in the same frame.
+      rAFTime - prevRAFTime > 0.1
```

Comment:
This particular change fixes a bug in Safari where we post a rAF from inside another rAF, and that rAF fires within the same frame. One of the tests I added covers this scenario.